internal/cloud/address_provider.go

```diff
@@ -24,42 +24,25 @@ import (
 )
 
 type AddressProvider struct {
-	dc    *DiscoveryClient
-	addrs []pubcluster.Address
+	dc *DiscoveryClient
 }
 
-func NewAddressProvider(dc *DiscoveryClient, addrs []Address) (*AddressProvider, error) {
-	if pubAddrs, err := translateAddrs(addrs); err != nil {
-		return nil, err
-	} else {
-		return &AddressProvider{
-			addrs: pubAddrs,
-			dc:    dc,
-		}, nil
-	}
-}
-
-func (a *AddressProvider) Addresses(refresh bool) []pubcluster.Address {
-	if refresh {
-		if err := a.refresh(); err != nil {
-			// TODO: log the error
-			return nil
-		}
-	}
-	return a.addrs
+func NewAddressProvider(dc *DiscoveryClient) *AddressProvider {
+	return &AddressProvider{dc: dc}
 }
 
-func (a *AddressProvider) refresh() error {
+func (a *AddressProvider) Addresses() []pubcluster.Address {
 	addrs, err := a.dc.DiscoverNodes(context.Background())
 	if err != nil {
-		return err
+		// TODO: log the error
+		return nil
 	}
 	pubAddrs, err := translateAddrs(addrs)
 	if err != nil {
-		return err
+		// TODO: log the error
+		return nil
```

Comment:
How about returning the error here and let the caller deal with it?

Reply:
Nice idea. Addressed in 48c7738f249c0e09edfbd62032e880f01f2c4666